File: apply-rls-migration.js
#!/usr/bin/env node

/**
 * Apply RLS migration to Supabase
 *
 * Usage:
 *   node apply-rls-migration.js <SERVICE_ROLE_KEY>
 *
 * Get your service role key from:
 *   https://app.supabase.com/project/lrytuvjouolizcgknqpd/settings/api
 */

const fs = require("fs");
const path = require("path");

const supabaseUrl = "https://lrytuvjouolizcgknqpd.supabase.co";
const serviceRoleKey = process.argv[2];

if (!serviceRoleKey) {
  console.error("❌ Error: SERVICE_ROLE_KEY not provided");
  console.error("");
  console.error("Usage: node apply-rls-migration.js <SERVICE_ROLE_KEY>");
  console.error("");
  console.error("Get your service role key from:");
  console.error(
    "  https://app.supabase.com/project/lrytuvjouolizcgknqpd/settings/api",
  );
  process.exit(1);
}

const sql = fs.readFileSync(
  path.join(__dirname, "supabase/migrations/003_add_rls_policies.sql"),
  "utf-8",
);

async function applyMigration() {
  try {
    console.log("📋 Applying RLS migration...");

    const response = await fetch(`${supabaseUrl}/rest/v1/rpc/exec_sql`, {
      method: "POST",
      headers: {
        Authorization: `Bearer ${serviceRoleKey}`,
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ sql }),
    });

    if (!response.ok) {
      const error = await response.text();
      throw new Error(`HTTP ${response.status}: ${error}`);
    }

    console.log("✅ RLS policies successfully applied!");
    console.log("");
    console.log("The following tables now have RLS policies:");
    console.log("  ✓ payroll_records");
    console.log("  ✓ deductions");
    console.log("  ✓ payslips");
    console.log("  ✓ tax_forms");
    console.log("  ✓ users");
    console.log("  ✓ departments");
    console.log("  ✓ audit_logs");
  } catch (error) {
    console.error("❌ Error applying migration:", error.message);
    process.exit(1);
  }
}

applyMigration();

File: apply_rls_migration.py
#!/usr/bin/env python3
"""
Apply RLS migration to Supabase

Usage:
    python3 apply_rls_migration.py <SERVICE_ROLE_KEY>

Get your service role key from:
    https://app.supabase.com/project/lrytuvjouolizcgknqpd/settings/api
"""

import sys
import json
from pathlib import Path
import subprocess

SUPABASE_URL = "https://lrytuvjouolizcgknqpd.supabase.co"

def apply_migration(service_role_key: str) -> bool:
    """Apply the RLS migration to Supabase"""
    
    # Read the migration SQL
    migration_path = Path(__file__).parent / "supabase" / "migrations" / "003_add_rls_policies.sql"
    
    if not migration_path.exists():
        print(f"❌ Migration file not found: {migration_path}")
        return False
    
    with open(migration_path, 'r') as f:
        sql = f.read()
    
    print("📋 Applying RLS migration...")
    print()
    
    # Prepare the curl command
    headers = {
        "Authorization": f"Bearer {service_role_key}",
        "Content-Type": "application/json",
    }
    
    payload = {
        "sql": sql
    }
    
    # Build curl command
    cmd = [
        "curl",
        "-X", "POST",
        f"{SUPABASE_URL}/rest/v1/rpc/exec_sql",
    ]
    
    for key, value in headers.items():
        cmd.extend(["-H", f"{key}: {value}"])
    
    cmd.extend([
        "-d", json.dumps(payload)
    ])
    
    # Execute curl
    try:
        result = subprocess.run(cmd, capture_output=True, text=True)
        
        if result.returncode == 0:
            print("✅ RLS policies successfully applied!")
            print()
            print("The following tables now have RLS policies:")
            print("  ✓ payroll_records")
            print("  ✓ deductions")
            print("  ✓ payslips")
            print("  ✓ tax_forms")
            print("  ✓ users")
            print("  ✓ departments")
            print("  ✓ audit_logs")
            return True
        else:
            print(f"❌ Error applying migration")
            print(f"Status: {result.returncode}")
            print(f"Error: {result.stderr}")
            return False
            
    except Exception as e:
        print(f"❌ Error: {e}")
        return False

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("❌ Error: SERVICE_ROLE_KEY not provided")
        print()
        print("Usage: python3 apply_rls_migration.py <SERVICE_ROLE_KEY>")
        print()
        print("Get your service role key from:")
        print("  https://app.supabase.com/project/lrytuvjouolizcgknqpd/settings/api")
        sys.exit(1)
    
    service_role_key = sys.argv[1]
    success = apply_migration(service_role_key)
    sys.exit(0 if success else 1)
